### ISSUE TITLE
Request: show warnings from go vet (and maybe gometalinter)

### ISSUE THREAD

**Issue by veqryn** [go-lang-plugin-org/go-lang-idea-plugin#2220]
It would be great if `go vet` was run in the background every few minutes, marking any errors or warnings it sees just like other inspections.
If this is supposed to be happening already, then it isn't for me (easy way to recreate this is to write a closure that captures a non-final variable, which gives a warning by go vet, which does not show up as a warning when viewing the source in intellij idea).
Another option would be to use `gometalinter` (https://github.com/alecthomas/gometalinter) with a customizable list of linters.

My system:
IDEA version 15.0.2
Go plugin version 0.10.749
Go version 1.5.2

**Comment by dlsniper** [comment]
Hi.

Unfortunately at the moment not all warnings that `go vet` can emit are translated into the plugin' inspections but this will happen in the next weeks / months (depending on how many contributions the plugin receives as well).

While it would be possible to have the warnings shown, see the previous: https://github.com/go-lang-plugin-org/go-lang-idea-plugin/pull/1554, it's not desirable as the IntelliJ platform can do much more afterwards (quickfixes for example). As for the `gometalinter` most of the functionality is either already implemented or there's an issue for the missing ones (if I've missed creating one then please open up a new issue).

If you would like to have any particular inspection showing up faster, you can pick one from this list https://github.com/go-lang-plugin-org/go-lang-idea-plugin/labels/inspections and try to contribute to the plugin.

I'll close this for now as this won't be implemented in the way desired in the request. Also please feel free to further open up issues with possible missing inspections or inspections that you'd like to see offered by the plugin.

Thank you.

**Comment by dlsniper** [comment]
> easy way to recreate this is to write a closure that captures a non-final variable

Can you please open up a new issue with this and a code sample to reproduce the issue? I'm not sure I understand it. Thank you.

**Comment by ignatov** [comment]
@zolotov Should we add support for such linters but turned off by default?

**Comment by ignatov** [comment]
Relates to #342

**Comment by dlsniper** [comment]
@ignatov in the discussion from
https://github.com/go-lang-plugin-org/go-lang-idea-plugin/pull/1554 the
outcome was that it's better to have all the checks translated to
inspections.

My plan is to provide at least one inspection per day in the coming weeks
so that we are on par with go vet, go lint and the other checks from
gometalinter (which would take about 2 months to finish). For instance, the
follow up ticket was a duplicate issue of one of the inspection I would
like to cover in the next weeks.

If you think it's ok to redo that PR I can have it reopen by the end of the
weekend.

What's the direction for this?

**Comment by daemonfire300** [comment]
I just installed and discovered this plugin today, coming from Atom (go-plus).
Wouldn't it be sufficient to just display the `go vet` and `golint` output on file change in a terminal like window at the bottom, just like when doing "vagrant up", which basically opens a terminal and runs the command?

**Comment by zolotov** [comment]
@daemonfire300 no, it wouldn't sufficient. You can do this with External Tools now
